core/middleware/index.js

```diff
@@ -1,6 +1,83 @@
 var util = require('util');
 
-var options = options;
+var options = opts;
+
+/**
+ * get object value safely
+ * @param {Object|Null|undefined} object
+ * @param {String|Number|Array} [key] object proterties keys
+ * @returns {*} result || null
+ */
+function getValueByObjectKeys(object, key){
+    var keyTypeof = typeof key,
+        i,
```

Comment:
Please follow nodejs best practises, and use duplicated var definition syntax

```
var keyTypeof = typeof key;
var i;
var iMax;
```

I will add same js hint config as in Source core repo.